File: src/2015/day01/src/test/java/ca/terrylockett/aoc2015/day01/TestDay01.java
package ca.terrylockett.aoc2015.day01;

import ca.terrylockett.aoccommon.resources.Resources;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.net.URISyntaxException;

import static org.junit.jupiter.api.Assertions.assertEquals;

class TestDay01 {

	static final String TEST_FILE_NAME = "test-input.txt";

	static String testFilePath = "";

	@BeforeAll
	static void setup() throws URISyntaxException {
		testFilePath = Resources.getInputFilePath(TEST_FILE_NAME).orElseThrow();
	}

	@Test
	void testDay01part1() throws IOException {
		assertEquals(-1, Day01Runner.part1(testFilePath));
	}

	@Test
	void testDay01part2() throws IOException {
		assertEquals(5, Day01Runner.part2(testFilePath));
	}
}

File: src/2024/day01/src/test/kotlin/ca/terrylockett/aoc2024/day01/TestDay01.kt
package ca.terrylockett.aoc2024.day01

import ca.terrylockett.aoccommon.resources.Resources
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class TestDay01 {

	@Test
	fun part1() {
		val inputFilePath = Resources.getInputFilePath("test-input.txt").orElseThrow()
		assertEquals(11, part01(inputFilePath))
	}

	@Test
	fun part2() {
		val inputFilePath = Resources.getInputFilePath("test-input.txt").orElseThrow()
		assertEquals(31, part02(inputFilePath))
	}
}
